homeassistant/components/roborock/button.py

```diff
@@ -0,0 +1,116 @@
+"""Support for Roborock button."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from roborock.roborock_typing import RoborockCommand
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import slugify
+
+from .const import DOMAIN
+from .coordinator import RoborockDataUpdateCoordinator
+from .device import RoborockEntity
+
+
+@dataclass
+class RoborockButtonDescriptionMixin:
+    """Define an entity description mixin for button entities."""
+
+    command: RoborockCommand
+    param: list | dict | None
+
+
+@dataclass
+class RoborockButtonDescription(
+    ButtonEntityDescription, RoborockButtonDescriptionMixin
+):
+    """Describes a Roborock button entity."""
+
+
+CONSUMABLE_BUTTON_DESCRIPTIONS = [
+    RoborockButtonDescription(
+        key="reset_sensor_consumable",
+        device_class=ButtonDeviceClass.UPDATE,
```

Comment:
I think this device class should not be used for a reset.

https://developers.home-assistant.io/docs/core/entity/button/#available-device-classes

Reply:
You are definitely right - will resolve